### PR TITLE
Clean up realmStore

### LINKED ISSUE #1312: [client] in UIStore, change selectedRealmEntityId to ...structureEntityId

### PR DESCRIPTION
**Pull request by cwastche**
Fixes #1312

### CODE REVIEW

File: client/src/hooks/store/useRealmStore.tsx [hunk 1]
```diff
@@ -1,18 +1,10 @@
...
 }
 
 export const createRealmStoreSlice = (set: any) => ({
-  realmId: undefined,
-  setRealmId: (realmId: ID) => set({ realmId }),
-  realmEntityId: 0,
-  setRealmEntityId: (realmEntityId: ID) => set({ realmEntityId }),
-  realmEntityIds: [],
-  setRealmEntityIds: (realmEntityIds: { realmEntityId: ID; realmId: ID }[]) => set({ realmEntityIds }),
+  structureEntityId: 0,
```
Comment (aymericdelab) [inline client/src/hooks/store/useRealmStore.tsx:8]: create a const UNDEFINED_STRUCTURE = 0 and use that everytime we set to 0

File: client/src/ui/utils/utils.tsx [hunk 2]
```diff
@@ -226,8 +226,8 @@ export const copyPlayerAddressToClipboard = (address: ContractAddress, name: str
     });
 };
 
-export const isRealmSelected = (realmEntityId: ID, structures: any) => {
-  const selectedStructure = structures?.find((structure: any) => structure?.entity_id === realmEntityId);
+export const isRealmSelected = (structureEntityId: ID, structures: any) => {
```
[the remark this answers is not in the document]
Comment (cwastche) [reply client/src/ui/utils/utils.tsx:229]: This function specifically checks to see if selected structure is a realm:
```
[...]
return selectedStructure?.category === "Realm";
```
So I think the name is correct

[the remark this answers is not in the document]
Comment (aymericdelab) [reply client/src/ui/utils/utils.tsx:229]: my bad

**Comment by aymericdelab** [comment]
@cwastche check https://github.com/BibliothecaDAO/eternum/blob/d6574bbf5140e069ea2974394352749727d23c3a/client/src/hooks/helpers/useStructureEntityId.tsx#L43

**Review by aymericdelab — APPROVED** [review]

**Pull request merged**